### ISSUE TITLE
`hardhat node` output has ascii escape characters when redirecting to a file

### ISSUE THREAD

**Issue by fvictorio** [nomiclabs/hardhat#467]
(I'm not really sure if they are called escape characters in this case, but you get the idea.)

How to reproduce:

```
hardhat node > log.txt
```

and then send a couple of repeated RPC calls, for example doing

```
curl -H "Content-Type: application/json" -X POST --data \
        '{"id":1337,"jsonrpc":"2.0","method":"eth_blockNumber","params":[]}' \
        http://localhost:8545
```

twice.

Result:

```
eth_blockNumber
[?25l[Feth_blockNumber (2)
[K[?25h
```

I think a `if (process.stdout.isTTY)` check somewhere should fix this.

**Comment by cameel** [comment]
I'm not completely sure but these look a bit like [ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) interpreted as text with control characters rather than colors (especially the `[`). If that's the case then it's somewhat related to #680 and https://github.com/ethereum/solidity/issues/11507.

**Comment by fvictorio** [comment]
I think this is related to our node logger, not to the compilation pipeline. The problem is that we use that special character to update the "current" line. We should do that only when stdout is not a tty.

FWIW: the solc CLI seems to handle this correctly as far as I can tell

**Comment by cameel** [comment]
> FWIW: the solc CLI seems to handle this correctly as far as I can tell

Yeah. That's because we do have a check that's basically equivalent to `if (process.stdout.isTTY)`: https://github.com/ethereum/solidity/blob/v0.8.6/solc/CommandLineInterface.cpp#L1130

**Comment by fvictorio** [comment]
Fixed in [`hardhat@2.8.0`](https://github.com/nomiclabs/hardhat/releases/tag/hardhat%402.8.0)
